src/coreclr/utilcode/stresslog.cpp

```diff
@@ -164,6 +164,11 @@ void AppendPid(LPCWSTR logFilename, LPWSTR fileName, size_t fileNameLength)
 
         // append the rest of the filename
         wcscat_s(fileName, fileNameLength, logFilename + pidInx + wcslen(pidLit));
+        return fileName;
+    }
+    else
+    {
+        return logFilename;
```

Comment:
The memory management rules of doing this feel error prone if sometimes we are returning a pointer to one buffer and other times we are returning a pointer to a different buffer. I'd suggest copying logFilename to fileName, make the return type void again, and the caller will always use fileName as before.

I don't think any of the callers that currently exist have an issue here, but I worry that it would be easy to create a new caller in the future that did have an issue.